frontend/src/metabase-types/api/settings.ts

```diff
@@ -49,6 +49,7 @@ export interface Settings {
   "show-homepage-pin-message": boolean;
   "show-lighthouse-illustration": boolean;
   "show-metabot": boolean;
+  "token-status": Record<string, unknown> | undefined;
```

Comment:
I know it's a draft, but please add a proper type for `TokenStatus` instead of `Record` when the PR is ready.

Reply:
And please also add `createMockTokenStatus` in `mocks/settings.ts` and use it in `createMockSettings` like other mocks are used

Reply:
I've made the `TokenStatus` type have as minimal constraints as possible, basically making it just for documentation purposes. I think we need to allow anything to be put in that object, if we want to allow store.metabase.com to change the shape of the token status response over time.

What do you think? Should I add more constraints?

https://github.com/metabase/metabase/blob/24bdd4fb3a124b358e229a026817183f8b493190/frontend/src/metabase-types/api/settings.ts#L35